mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/RuntimeTaskImpl.java

```diff
@@ -99,8 +99,9 @@ public void initialize(
             this.wrappedExecuteStageRequest =
                 new WrappedExecuteStageRequest(PublishSubject.create(), executeStageRequest);
 
-            log.info("Picking Cgroups metrics collector.");
-            configWritable.setMetricsCollector(CgroupsMetricsCollector.valueOf(System.getProperties()));
+            MetricsCollector metricsCollector = config.getUsageSupplier();
```

Comment:
hi @arnavdugar we are trying to move away from passing objects via the config class and this config class you use here is generated from the sterilized version (L94 -96) so the supplier is always empty.
If you need to support another metrics collector I would suggest adding a config value as string and then construct the metrics collector type instance based the string value instead (basically treat the current interface at initialize as entry and do not pass external instances from there). We need this assumption to support our internal wrapper to these entries.